src/api/app/models/workflow/step/configure_repositories.rb

```diff
@@ -64,4 +68,15 @@ def validate_project_name
 
     errors.add(:base, "invalid project '#{target_project_base_name}'")
   end
+
+  def write_report
+    return unless workflow_run = WorkflowRun.find(workflow_run_id)
```

Comment:
Looks to me like you want to dry this up in `Workflow::Step`. The only difference between the step types is the json content right?

Would also greatly simplify the specs you need to touch etc.

Reply:
I did some refactoring already. I moved the `write_report` to the Step model and in every specific step, I simply define the method `artifacts` that returns only the hash with the information.